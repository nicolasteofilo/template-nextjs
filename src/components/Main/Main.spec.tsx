import { screen, render } from '@testing-library/react'
import { Main } from '.'

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />)

    expect(
      screen.getByRole('heading', {
        name: /Typescript, ReactJS, NextJS and Styled Components Template/i
      })
    ).toBeInTheDocument()
  }),
    it('should render correct', () => {
      const { container } = render(<Main />)

      expect(container.firstChild).toMatchSnapshot()
    })
  it('should render the colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({
      'background-color': '#06092b'
    })
  })
})
