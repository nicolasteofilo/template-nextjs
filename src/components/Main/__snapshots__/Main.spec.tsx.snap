// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render correct 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  height: 100%;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 code {
  font-style: italic;
  background-color: #666666;
  padding: 0.2rem;
  border-radius: 5px;
}

.c1 {
  width: 15rem;
  margin: 0 auto;
  margin-bottom: 20px;
}

.c2 {
  font-size: 2rem;
  font-weight: normal;
}

.c2 span {
  font-weight: bold;
}

.c3 {
  font-size: 1.5rem;
  margin-top: 1rem;
  margin-bottom: 1rem;
}

<main
  class="c0"
>
  <img
    alt="Logo do frameword nextjs e escrito NextJs na PrÃ¡tica ao lado"
    class="c1"
    src="/img/nextjs.png"
  />
  <h2
    class="c2"
  >
    Typescript, ReactJS, NextJS and Styled Components
    <span>
       Template
    </span>
  </h2>
  <p
    class="c3"
  >
    Edit 
    <code>
      src/components/Main/index.tsx
    </code>
     and save to update.
  </p>
</main>
`;
